src/main/java/net/spy/memcached/MemcachedConnection.java

```diff
@@ -257,6 +257,7 @@ public class MemcachedConnection extends SpyThread implements ClusterConfigurati
   //of node list to the locator object.
   protected final ReentrantLock conditionLock;
   protected final Condition nodeUpdateCondition;
+  protected boolean isNodeUpdateConditionInClientInit = false;
```

Comment:
I would rename this to be something like `isInitialClusterConfigApplied`. For the start-up case you are only interested in whether or not the initial configuration has been applied or not.